### DISCUSSION TITLE
Formula `freetype`'s pkg-config file direct to old `libpng` library

### DISCUSSION THREAD

**Discussion by ywwry66** [Homebrew/discussions#3715] [Everyday usage] [answered]
In the pkg-config file `freetype2.pc`, there are the lines:
```
Requires.private: 
Libs: -L/opt/homebrew/opt/freetype/lib -lfreetype
Libs.private: -lz -lbz2 -L/opt/homebrew/Cellar/libpng/1.6.37/lib -lpng16 -lz
Cflags: -I${includedir}/freetype2
```
The latest version of `libpng` is 1.6.38.

If I reinstall from source, it has the following:
```
Requires.private: zlib, libpng
Libs: -L/opt/homebrew/opt/freetype/lib -lfreetype
Libs.private: -lbz2
Cflags: -I${includedir}/freetype2
```
This works well since it directs to `libpng.pc` file.

**Comment by ywwry66** [comment, marked as answer]
I also observe the same behavior on my other Mac. Why the CI built `pkg-config` file has hard coded absolute path in it while local built file does not? Does this mean formula `freetype` needs a revision bump when `libpng` get updated?

Edit: See the discussions below.

Edit2: See https://github.com/Homebrew/homebrew-core/pull/124287

**Reply by SMillerDev** [reply]
Currently, yes. But a better fix would be to stop the `freetype` formula from hardcoding that path.

**Reply by ywwry66** [reply]
Thanks for the answer. Do you know what caused the difference that when I built `freetype` from source locally using `brew reinstall -s freetype`, the pc file does not include the absolute path?

**Reply by SMillerDev** [reply]
Maybe a brew change that was made after the last bottle of freetype

**Reply by ywwry66** [reply]
That makes sense. Thank you.
